service/controller/app/resource/status/create.go

```diff
@@ -47,6 +41,12 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			},
 		}
 	} else {
+		if cc.Status.ClusterStatus.IsUnavailable {
```

Comment:
I moved these statements here since we need to update the app CR status even when the tenant cluster is not available.